### PR TITLE
Feature/discussion cards

### PR DESCRIPTION
**Pull request by bgmad**
## PR TITLE

**Description:** 
updated discussion cards. Also refactored the structure for displaying the cards. Both RoomContent.js and DashboardContent.js (eventually) should use the Feed.js component to display discussion cards.

### CODE REVIEW

File: src/assets/DiscussionCard.js [hunk 1]
```diff
@@ -1,91 +1,113 @@
...
+      {props.item.flags && (
+        <Link
+          style={{
+            width: '100%',
+            display: 'flex',
+            justifyContent: 'flex-end',
+          }}
+          to={`${url}/discussion/${props.item.id}?view=flagged`}
+        >
+          <FlagChip flags={props.item.flags.length} />
```
Comment (mxhs) [inline src/assets/DiscussionCard.js:101]: Yay chip !

File: src/assets/PopoverContent.js [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+            Mod
+          </Button>
+        </Link>
+        <FlagManagerModal
+          visible={showModal}
+          setVisible={setShowModal}
+          flagsData={props.discussion.flags || null}
+          discussionID={props.discussion.id}
+        />
+      </List.Item> */}
```
Comment (mxhs) [inline src/assets/PopoverContent.js:55]: Hmm any reason why the manager modal is commented out? Is this the pseudo-temp button we talked about?

File: src/assets/Feed.js [hunk 3]
```diff
@@ -1,13 +1,59 @@
...
-  return <h2>Feed</h2>;
+import DiscussionCard from './DiscussionCard';
+
+const Feed = (props) => {
+  const { roomID } = useParams();
+
+  useEffect(() => {
+    if (roomID) {
+      if (props.user.role_id < 2) props.fetchPostByRoom(roomID, 1);
+      else props.fetchPostsAndFlagsByRoom(roomID, 1);
```
Comment (mxhs) [inline src/assets/Feed.js:18]: ahh conditional logic ayeeeee. Good catch, but middleware and stuff is cool too echem. (Will be good also for when future cohorts implement the permissions table logic)

**Review by mxhs — APPROVED** [review]
Nothing major. Looks good, just might be good to use the helper component to check user permissions so its malleable for future cohorts.

**Review by CPower1248 — APPROVED** [review]

**Pull request merged**